cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -224,11 +228,14 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         private ConfigMap topicOperatorMetricsAndLogsConfigMap = null;
         private ConfigMap userOperatorMetricsAndLogsConfigMap;
 
+        private List<String> maintenanceTimeWindows;
```

Comment:
Genuine question, what's the advantage of having `kafkaAssembly.getSpec().getMaintenanceTimeWindows()` inside a method, if every time we call that method it means to call always two methods on `kafkaAssembly`? I would see to call it one time and saving a reference to the returned list (as happens now) but maybe declaring `maintenanceTimeWindows` as a local variable in the `isMaintenanceTimeWindowsSatisfied` where it's used. It's also true that it means calling `kafkaAssembly.getSpec().getMaintenanceTimeWindows()` more time in a single rinconcile while it's enough doing that just one time, for this reason the member field. Putting this in a method seems to be just a way to "hide" what's happening underneath for having more clean code.

Reply:
So I am a little bit confused now. While thanks to inlining by JIT compiler we don't have performance issues, we have them in this case because each method call means contacting the apiserver. So it seems saving in a variable seems to be better, having the call just one time. What am I missing or I did miss understand from what you said?

Reply:
> we have them in this case because each method call means contacting the apiserver

You misunderstand. `kafkaAssembly` is already a field of `ReconciliationState`, so I'm saying there's no need to add this field as well. The value can be obtained from `kafkaAssembly.getSpec().getMaintenanceTimeWindows()` which doesn't involve the the apiserver at all. It's just some getters on a `Kafka` that's already been fetched from the apiserver. If you only need that value once just use `kafkaAssembly.getSpec().getMaintenanceTimeWindows()`. If you need it in multiple places (which I don't think you do) I'm saying you can put `kafkaAssembly.getSpec().getMaintenanceTimeWindows()` in a method to remove the duplicate code.

Reply:
Ok then, but it seems that a method is needed because it's used 3 times in the same method : 

```
private boolean isMaintenanceTimeWindowsSatisfied() {
            try {
                boolean isSatisfiedBy = maintenanceTimeWindows == null || maintenanceTimeWindows.isEmpty();
                if (!isSatisfiedBy) {
                    Date date = new Date();
                    for (String cron : maintenanceTimeWindows) {
...
...

```